java/core/src/main/java/co/worklytics/psoxy/PseudonymizerImpl.java

```diff
@@ -0,0 +1,148 @@
+package co.worklytics.psoxy;
+
+import com.avaulta.gateway.pseudonyms.Pseudonym;
+import com.avaulta.gateway.pseudonyms.PseudonymImplementation;
+import com.avaulta.gateway.pseudonyms.impl.UrlSafeTokenPseudonymEncoder;
+import com.avaulta.gateway.rules.transforms.Transform;
+import com.avaulta.gateway.tokens.DeterministicTokenizationStrategy;
+import com.avaulta.gateway.tokens.ReversibleTokenizationStrategy;
+import com.google.common.base.Preconditions;
+import dagger.assisted.Assisted;
+import dagger.assisted.AssistedInject;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.java.Log;
+import org.hazlewood.connor.bottema.emailaddress.EmailAddressCriteria;
+import org.hazlewood.connor.bottema.emailaddress.EmailAddressParser;
+import org.hazlewood.connor.bottema.emailaddress.EmailAddressValidator;
+
+import javax.inject.Inject;
+import java.util.Objects;
+import java.util.function.Function;
+
+@NoArgsConstructor
+@Log
+public class PseudonymizerImpl implements Pseudonymizer {
+
+    @Inject
+    HashUtils hashUtils;
+
+    @Inject
+    ReversibleTokenizationStrategy reversibleTokenizationStrategy;
+    @Inject
+    DeterministicTokenizationStrategy deterministicTokenizationStrategy;
+    @Inject
+    UrlSafeTokenPseudonymEncoder urlSafePseudonymEncoder;
+
+    @Getter
+    ConfigurationOptions options;
+
+    @AssistedInject
+    public PseudonymizerImpl(@Assisted Pseudonymizer.ConfigurationOptions options) {
+        this.options = options;
+    }
+
+    @Override
+    public PseudonymizedIdentity pseudonymize(@NonNull String value) {
+        return pseudonymize((Object)  value);
+    }
+
+    @Override
+    public PseudonymizedIdentity pseudonymize(@NonNull Number value) {
+        return pseudonymize((Object) value);
+    }
```

Comment:
really needed in the interface? they just call the Object one. That could deal internally with specialized versions per object class.

Reply:
yeah, maybe can be eliminated. The `Object` one was previously private